daemon/container_operations_unix.go

```diff
@@ -622,28 +652,14 @@ func (daemon *Daemon) ConnectToNetwork(container *container.Container, idOrName
 }
 
 func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName string, updateSettings bool) (err error) {
-	if container.HostConfig.NetworkMode.IsContainer() {
-		return runconfig.ErrConflictSharedNetwork
-	}
-
-	if containertypes.NetworkMode(idOrName).IsBridge() &&
-		daemon.configStore.DisableBridge {
-		container.Config.NetworkDisabled = true
-		return nil
-	}
-
-	controller := daemon.netController
-
-	n, err := daemon.FindNetwork(idOrName)
+	n, err := daemon.updateNetworks(container, idOrName, updateSettings)
```

Comment:
That's not true. please note the subtle difference in the function name (I should be blamed there).
`ConnectToNetwork` is called from `network connect` UX. while `connectToNetwork` is a common function called both from `ConnectToNetwork` and `AllocateNetwork` (which is the entry point for `docker run --net`).

So we cannot remove this call from here (to handle the `docker run` cases).

Hence, we have to be careful in doing the change that you suggested above https://github.com/docker/docker/pull/18906/files#r49092752.